lib/demo-block-templates/front-page.html

```diff
@@ -1,3 +1,4 @@
+<!-- wp:template-part {"slug":"header","theme":"twentytwenty"} /-->
```

Comment:
Do we need the theme slug here? It'd be good if it could be implicit to the "current theme" as set in options.

Reply:
It's so that customized template parts can persist after switching themes. Different themes could both define headers.

Do you mean to make it default to the current theme if not set? We can do that, but we would need to save it on customized templates so that they don't load a different theme's template parts after a theme switch. It seems like a lot of additional implicit behaviors that might confuse people. I'm not sure saving one attribute in theme files is worth it.

Reply:
It's a weird requirement to have just for the sake of switching themes. It should be possible to handle it at the time of switching themes as well (also depending on what the UX looks like for what you want to preserve and what you want to replace) since we know both the previous and after contexts. Also not even sure that we want the concept of mixed parts between themes, I'd lean more towards a split of just `current theme` and `user templates`.

Reply:
>It should be possible to handle it at the time of switching themes as well (also depending on what the UX looks like for what you want to preserve and what you want to replace) since we know both the previous and after contexts.

That is the implicit behavior I mentioned could be confusing.

>Also not even sure that we want the concept of mixed parts between themes, I'd lean more towards a split of just `current theme` and `user templates`.

It would be confusing if, after switching themes, those `user templates` started referencing different template parts, hence why we require the attribute after customization.

We could default to the theme that originally created each `user template`, but then how does the user add new template parts to existing `user templates`.

Reply:
I also didn't expect to see the theme slug here, since I interpreted these demo templates as a generic fallback that should work with any given theme.

> It's so that customized template parts can persist after switching themes. Different themes could both define headers.

Having said that, I do agree with this part in general. It just didn't seem necessary to me in this particular case. Would it make sense to persist it only once the demo template part has been modified?

Reply:
What if the user modifies the template, without touching the part, switches themes, and then ends up with a different template part reference. There are too many edge cases like this where we lose with nothing to win.